src/encord_active/app/data_quality/sub_pages/explorer.py

```diff
@@ -131,17 +131,26 @@ def build(self, selected_df: DataFrame[MetricSchema], metric_scope: MetricScope)
         fill_data_quality_window(selected_df, metric_scope, selected_metric)
 
 
+class PointSelectionSchema(pa.SchemaModel):
+    x: Series[float] = pa.Field(coerce=True)
+    y: Series[float] = pa.Field(coerce=True)
+    curveNumber: Series[float] = pa.Field(coerce=True)
+    pointNumber: Series[float] = pa.Field(coerce=True)
+    pointIndex: Series[float] = pa.Field(coerce=True)
+
+
 def get_selected_rows(
     embeddings_2d: DataFrame[Embedding2DSchema], selected_points: list[dict]
 ) -> DataFrame[Embedding2DSchema]:
-    """
-    Selected points from the plotly graph only have x and y values, so we need a hacky way to identify selected points
-    Here, we just create and additional column by merging x and y points, so that we can get the selected points'
-    identifiers for filtering.
-    """
-    selected_points_merged = [str(int(item["x"] * 1000)) + "-" + str(int(item["y"] * 1000)) for item in selected_points]
-
-    selected_rows = embeddings_2d[embeddings_2d[Embedding2DSchema.x_y].isin(selected_points_merged)]
+    selection_raw = DataFrame[PointSelectionSchema](pd.DataFrame(selected_points))
+    matched = embeddings_2d.copy().merge(
+        selection_raw[[PointSelectionSchema.x, PointSelectionSchema.y]],
+        on=[Embedding2DSchema.x, Embedding2DSchema.y],
```

Comment:
Since we use different schemas here for the x and y points, what about creating a new point schema as a parent class for both of them here to avoid future bugs:

```
class PointSchema(pa.SchemaModel):
    x: Series[float] = pa.Field(coerce=True)
    y: Series[float] = pa.Field(coerce=True)

class PointSelectionSchema(PointSchema):
    curveNumber: Series[float] = pa.Field(coerce=True)
    pointNumber: Series[float] = pa.Field(coerce=True)
    pointIndex: Series[float] = pa.Field(coerce=True)
```